Assets/Code/src/Runtime/Match/Components/MatchCountdownUI.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Collections;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+namespace HouraiTeahouse.FantasyCrescendo.Matches
+{
+
+public class MatchCountdownUI : MonoBehaviour
+{
+    // Component References
+    // EditorBrowsable attribute hides the public variables from intellisense
+    // In other words, you can still access it, but it won't be in the autocomplete list
+    // If variables are null, look for them
+    [Header("Component References")]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public TextMeshProUGUI _TextUI;
+    public TextMeshProUGUI TextUI {
+        get {
+            if (!_TextUI)
+                _TextUI = GetComponentInChildren<TextMeshProUGUI>();
+            return _TextUI;
+            }
+        }
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public AudioSource _AudioPlayer;
+    public AudioSource AudioPlayer
+    {
+        get
+        {
+            if (!_AudioPlayer)
+                _AudioPlayer = GetComponentInChildren<AudioSource>();
+            return _AudioPlayer;
+        }
+    }
+
+    [Header("Timings")]
+    public float InitialDelay = 1f;
+    public float GoDuration = 1.25f;
+
+    [Header("Countdown Clips (0 = 'GO', 1-10 = Time)")]
+    public AudioClip[] CountdownClips;
+
+    [Header("Debug")]
+    public bool DisableCountdown = false;
+
+    /// <summary>
+    /// Awake is called when the script instance is being loaded.
+    /// </summary>
+    void Awake()
+    {
+        Mediator.Global.CreateUnityContext(this).Subscribe<MatchStartCountdownEvent>(StartCountdown);
+    }
+
+    /// <summary>
+    /// Starts countdown UI. Counts from given starting number to 1 to GO!
+    /// Returns when GO! is reached, but GO! will continue to be displayed for a short duration
+    /// </summary>
+    /// <param name="evt"></param>
+    /// <returns></returns>
+    async Task StartCountdown(MatchStartCountdownEvent evt)
+    {
+        // For Debug purposes
+        if (DisableCountdown) return;
```

Comment:
To make sure that this does not affect non-debug builds, boolean AND with `Debug.isDebugBuild`